aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1006,10 +1006,10 @@ Tensor multi_dot_impl(TensorList _tensors, c10::optional<Tensor> _out) {
     // output has shape (a, c). If either the first or last tensor is 1D
     // a and/or c dimensions will be implicitely size 1 and will be ommited
     // from the output. e.g. for inputs (a, b) x (b) the output has shape (a,).
-    at::native::resize_output(out, out_shape);
+    at::native::resize_output_symint(out, out_shape);
 
     // View output as 2D for simplicity of computation.
-    result = out.view({tensors[0].size(0), tensors.back().size(-1)});
+    result = out.view_symint({tensors[0].sym_size(0), tensors.back().size(-1)});
```

Comment:
Perhaps the issue with out is caused by returning `result` rather than `out`? Otherwise, I'd like to understand why is it failing, as I don't see the reason why it should...